### PR TITLE
Fix handling of aot namespace config

### PR DESCRIPTION
**Pull request by tux-rampage**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

When the AoT namespace is an empty array, it will bypass an expected exception that should reveal a misconfiguration issue.
This was revealed by a failing test case.

**This failing test only affects the `develop`branch.**

See #11 and #10

### CODE REVIEW

File: src/GeneratedInjectorDelegator.php [hunk 1]
```diff
@@ -22,7 +22,9 @@ public function __invoke(ContainerInterface $container, $name, callable $callbac
     {
         $config = $container->has('config') ? $container->get('config') : [];
         $aotConfig = $config['dependencies']['auto']['aot'] ?? [];
-        $namespace = empty($aotConfig['namespace']) ? 'Laminas\Di\Generated' : $aotConfig['namespace'];
+        $namespace = ! isset($aotConfig['namespace']) || $aotConfig['namespace'] === ''
```
Comment (tux-rampage) [inline src/GeneratedInjectorDelegator.php:25]: An empty array is a result of a potential misconfiguration. The `empty()` check will swallow this and not inform the user about it. Using the null coalescing check would not catch an empty string. Therefore `isset()` and explicit empty string check.

**Review by michalbundyra — COMMENTED** [review]
That's a bit tricky... Nobody wants run composer update and got fatal after...

I see it was wrong configuration, so in general we should blame users for that, but... those who are gonna have broken system are going to blame us...

Anyway, I would say GO FOR IT, maybe add a bit more to CHANGELOG, so people knows it's potential BC Break in case of misconfiguration.

File: src/GeneratedInjectorDelegator.php [hunk 2]
```diff
@@ -22,7 +22,9 @@ public function __invoke(ContainerInterface $container, $name, callable $callbac
...
+            ? 'Laminas\Di\Generated'
+            : $aotConfig['namespace'];
```
Comment (Ocramius) [inline src/GeneratedInjectorDelegator.php:27]: Is there an effective way to test this behavior? 

Also, code above can probably be rewritten as `$namespace = ($aotConfig['namespace'] ?? 'Laminas\Di\Generated') ?: 'Laminas\Di\Generated';`, but that's a matter of taste (mostly reducing number of AST nodes)

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/GeneratedInjectorDelegator.php (hunk 2 above)
Comment (tux-rampage) [reply src/GeneratedInjectorDelegator.php:27]: Thanks for taking time to review.

> Is there an effective way to test this behavior?

There already is a (failing) Testcase which this PR is attempting to fix or am I missing something :thinking: ?

> Also, code above can probably be rewritten as (...)

Sweet :+1:

**Comment by tux-rampage** [comment]
@michalbundyra Thanks for reviewing :+1:

I'll rephrase the Changelog. Could rephrasing the Exception message with a Link to this PR be helpful?

**Comment by tux-rampage** [comment]
@michalbundyra This is affecting a fix for a pending feature introduced with https://github.com/zendframework/zend-di/pull/51 - which is not yet released. Therefore no one should be affected unless they explicitly require `^3.2@dev` or `dev-develop` - In this case I'd consider users to be aware of potential breaks.

Maybe we can drop the Changelog-Entry entirely? What do you think?

File: src/GeneratedInjectorDelegator.php (hunk 2 above)
Comment (tux-rampage) [reply src/GeneratedInjectorDelegator.php:27]: @Ocramius Sorry, the proposed `$namespace = ($aotConfig['namespace'] ?? 'Laminas\Di\Generated') ?: 'Laminas\Di\Generated';` seems not to work as expected:

The ternary expression (`?: 'Laminas\Di\Generated`) catches the empty array just like `empty()` before, so the test keeps failing.

Comment (Ocramius) [reply src/GeneratedInjectorDelegator.php:27]: Oh, didn't know this could be an `array`, sorry...

**Review by Ocramius — APPROVED** [review]

**Comment by michalbundyra** [comment]
@tux-rampage yes, if the issue is present only on develop we can drop the changelog (unless some behaviour has been changed previously and it was not noted in changelog?)

**Comment by tux-rampage** [comment]
> ... (unless some behaviour has been changed previously and it was not noted in changelog?)

No this is not the case. It was introduced with a completely new feature (new delegator factory that did not exist before). So this actually finishes the feature from PR zendframework/zend-di#51 listed in the [changelog](https://github.com/laminas/laminas-di/blob/develop/CHANGELOG.md#320---tbd) under 3.2.0 (currently TBD).

**Review by michalbundyra — APPROVED** [review]
🚢

**Pull request merged**